src/shared/shared.module.ts

```diff
@@ -4,12 +4,28 @@ import { SvgIconComponent } from 'src/shared/components/svg-icon/svg-icon.compon
 import { CommonModule } from '@angular/common';
 import { NotFoundPageComponent } from 'src/shared/components/not-found-page/not-found-page.component';
 import { SharedRoutingModule } from 'src/shared/shared-routing.module';
+import { MatDialogModule } from '@angular/material/dialog';
+import { MatDialogComponent } from 'src/shared/components/mat-dialog/mat-dialog.component';
+import { MatButtonModule } from '@angular/material/button';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { FormsModule } from '@angular/forms';
+import { MatInputModule } from '@angular/material/input';
 
 @NgModule({
-  declarations: [HeaderComponent, SvgIconComponent, NotFoundPageComponent],
+  declarations: [
+    HeaderComponent,
+    SvgIconComponent,
+    NotFoundPageComponent,
+    MatDialogComponent,
+  ],
   imports: [
     CommonModule,
     SharedRoutingModule,
+    MatDialogModule,
```

Comment:
If you add MatDialogModule, MatButtonModule,
    MatFormFieldModule,
    FormsModule,
    MatInputModule 

in SharedModule then if you will be import SharedModule in other module => you will be loading all from this modules

Reply:
aproove